heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java

```diff
@@ -0,0 +1,153 @@
+//  Copyright 2017 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.apiserver.resources;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.UUID;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.twitter.heron.apiserver.utils.Logging;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Key;
+
+/**
+ * Endpoints that allows the API server to become a file server
+ */
+@Path("/file")
+public class FileResource extends HeronResource {
+  private static final Logger LOG = LoggerFactory.getLogger(FileResource.class);
+
+  private static final String FILE_SYSTEM_DIRECTORY
+      = "heron.apiserver.http.file.system.directory";
+
+  private static InetAddress ip;
+  private static String hostname;
+
+  static {
+    try {
+      ip = InetAddress.getLocalHost();
+      hostname = ip.getHostName();
+    } catch (UnknownHostException e) {
+      LOG.info("Failed or resolve IP address of localhost");
+    }
+  }
+
+  @POST
+  @Path("/upload")
+  @Consumes(MediaType.MULTIPART_FORM_DATA)
+  public Response uploadFile(
+      @FormDataParam("file") InputStream uploadedInputStream,
+      @FormDataParam("file") FormDataContentDisposition fileDetail) {
+
+    Config config = createConfig();
+
+    if (uploadedInputStream == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
```

Comment:
Can we add some useful error messages here? There are some private helper methods (that could be refactored out) in the TopologyResources to help with this.